src/logic/operate.js

```diff
@@ -1,20 +1,27 @@
-import 'big.js';
-var Operate = (function () {
-
-  let mod = {};
-  mod.operate = function (calcObj, button) {
-    let { total, next, operation } = calcObj;
-    if (button === '+/1') {
-      total *= -1;
-      next *= -1;
-    }
-    return { total, next, operation }
+import Big from 'big.js';
 
+export default function operate(numbernumber1, numbernumber2, operation) {
+  let total = '';
+  const number1 = Big(numbernumber1);
+  const number2 = Big(numbernumber2);
+  switch (operation) {
+    case '+':
+      total = number1.plus(number2).toString();
+      break;
+    case '-':
+      total = number1.minus(number2).toString();
+      break;
+    case 'x':
+      total = number1.times(number2).toString();
+      break;
+    case '/':
+      total = number1.div(number2).toString();
+      break;
```

Comment:
This will cause your app to break, dividing by `0` will break the app. check if the second number is `0` and return error message.

Reply:
thanks for the observation